worktree_status.go

```diff
@@ -137,13 +138,13 @@ func (w *Worktree) diffStagingWithWorktree(reverse, excludeIgnoredChanges bool)
 		return nil, err
 	}
 
-	from := mindex.NewRootNode(idx)
-	submodules, err := w.getSubmodulesStatus()
+	cfg, err := w.r.ConfigScoped(config.SystemScope)
```

Comment:
This may lead to regression. Instead, I'd rather we add a  `IgnoreFileMode` (or `upholdFileMode`) to the `StatusOptions`.

This way, users can call `ConfigScoped(config.SystemScope)` and based on its result set that value - this will also fix the test failures.